src/components/Map/index.tsx

```diff
@@ -1,18 +1,62 @@
-import Map from "react-map-gl/maplibre";
+"use client";
 import mapStyle from "./mapStyle";
-import { FC } from "react";
+import { FC, useRef } from "react";
+import maplibregl from "maplibre-gl";
+import Map, { Source, Layer, MapLayerMouseEvent } from "react-map-gl/maplibre";
+import { layerStyles } from "./layerStyles";
 
-interface Props {}
-export const MapComponent: FC<Props> = () => {
+interface Props {
+  mapData: any;
+  activeDistrictSet: (label: string) => void;
+}
+
+const districtIDs = [
+  "district_1",
+  "district_2",
+  "district_3",
+  "district_4",
+  "district_5",
+];
+
+export const MapComponent: FC<Props> = ({ mapData, activeDistrictSet }) => {
+  const mapRef = useRef<mapboxgl.Map>();
+  const onMapCLick = (event: MapLayerMouseEvent) => {
+    event.preventDefault();
+    if (!mapRef || !mapRef.current) {
+      return;
+    }
+    /* @ts-ignore */
+    const districtName = mapRef.current.queryRenderedFeatures(event.point, {
+      layers: districtIDs,
+    })[0]?.properties?.name;
+    console.log(districtName);
+    activeDistrictSet(districtName);
+  };
+  const districtsFromGEO = JSON.parse(mapData.value).berlinDistricts.features;
   return (
     <Map
       initialViewState={{
         longitude: 13.41,
         latitude: 52.52,
-        zoom: 12,
+        zoom: 8,
       }}
+      maxBounds={[13.0, 52.3, 13.7, 52.7]}
+      // @ts-ignore
       mapStyle={mapStyle()}
```

Comment:
The map is flashing because this `mapStyle` is reset on every render (and that probably triggers some kind of reinitialization of the Map component). You can memoize it before to prevent that and only create this style object once:

```typescript
const memoizedMapStyle = useMemo(() => mapStyle(), []);
```

… and then change this line accordingly:

```suggestion
      mapStyle={memoizedMapStyle}
```

Reply:
Nice!